src/components/Comment/Comment.tsx

```diff
@@ -161,19 +141,56 @@ const ReplyArrow = () => (
   </svg>
 );
 
-export const Comment = ({
-  baseUrl,
+const ProfilName = ({ pillar, userId, displayName }: { pillar: Pillar, userId: string, displayName: string }) => (
+  <div className={commentProfileName(pillar)}>
+    <a
+      href={joinUrl(["https://profile.theguardian.com/user", userId])}
+      className={linkStyles}
+    >
+      {displayName}
+    </a>
+  </div>
+)
+
+const Badges = ({ comment, isHighlighted }: { comment: CommentType, isHighlighted: boolean }) => (
+  <>
+    <>
+      {!!comment.userProfile.badge.filter(
+        obj => obj["name"] === "Staff"
+      ).length && (
+          <div className={iconWrapper}>
+            <GuardianStaff />
+          </div>
+        )}
+    </>
+    <>
+      {comment.status !== "blocked" && isHighlighted && (
+        <div className={iconWrapper}>
+          <GuardianPick />
+        </div>
+      )}
+    </>
+  </>
+)
+
+const CommentMessage = ({
   comment,
   pillar,
   setCommentBeingRepliedTo,
   user,
-  isReply
-}: Props) => {
+  isHighlighted,
+  setIsHighlighted,
+  setError
+}: {
+  comment: CommentType,
+  pillar: Pillar,
+  setCommentBeingRepliedTo: (commentBeingRepliedTo?: CommentType) => void,
+  user?: UserProfile,
+  isHighlighted: boolean,
+  setIsHighlighted: (isHighlighted: boolean) => void,
+  setError: (errorMessage: string) => void
+}) => {
   const commentControlsButtonStyles = commentControlsButton(pillar);
```

Comment:
I know this is outside this PR but we could maybe remove this abstraction?

Reply:
good point